### PR TITLE
Changes suggested by readability-simplify-boolean-expr.

### PR DESCRIPTION
**Pull request by quantumsteve**
I ran clang-tidy's [readability-simplify-boolean-expr](http://clang.llvm.org/extra/clang-tidy/checks/readability-simplify-boolean-expr.html) check on Mantid Framework, letting it make changes. It found about 100 redundant boolean expressions. 

No release notes

testing: None of these changes should change behavior.

code review: I like that these changes eliminate a branch and therefore reduce cyclomatic complexity. Please flag any changes that make the code much more difficult to read.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
Is this really needed for release 3.6?

File: Framework/Geometry/src/Math/Acomp.cpp [hunk 1]
```diff
@@ -192,10 +188,8 @@ Order (low first)
     if (*ax != *ux)
       return (*ux < *ax);
   }
-  if (uc != Units.end())
-    return true;
-  // everything idential or A.comp bigger:
-  return false;
+  return uc != Units.end();
+  // false = everything idential or A.comp bigger:
```
Comment (SimonHeybrock) [inline Framework/Geometry/src/Math/Acomp.cpp]: Update comment?

**Comment by SimonHeybrock** [comment]
I like all changes, this increases readability significantly.

As @KarlPalmen mentioned, we can probably merge this right after the release, so we don't interfere with testing.

**Comment by quantumsteve** [comment]
Now that we've created a 3.6 release branch, is this ready to merge into master?

**Pull request merged**